apps/src/sites/code.org/pages/public/teacher-dashboard/index.js

```diff
@@ -366,7 +376,15 @@ function main() {
           $.each(resultStudents, function (index, student) {
             $scope.section.students.unshift(student);
           });
-        }).$promise.catch($scope.genericError);
+        }).$promise.then(() => {
+          // If we went from zero to more than zero students, rerender login
+          // type controls so the correct options are available.
+          // TODO: Once everything is React this should become unnecessary.
+          if (newStudents.length === $scope.section.students.length) {
+            unmountLoginTypeControls();
+            renderLoginTypeControls($scope.section.id);
```

Comment:
I don't understand how the comment describes this code. The code tells me that we rerender if newStudents length is the same as what I assume to be oldStudents length?